kork-credentials-api/src/main/java/com/netflix/spinnaker/credentials/definition/CredentialsDefinitionRepository.java

```diff
@@ -0,0 +1,152 @@
+   *
+   * @param typeName credential type to search for
+   * @return list of all stored credentials definitions matching the given account type name
+   */
+  List<? extends CredentialsDefinition> listByType(String typeName);
+
+  /**
+   * Lists credentials views for a given credentials type accessible to the current user.
+   *
+   * @return list of accessible credentials and their statuses
+   */
+  List<CredentialsView> listCredentialsViews(String typeName);
+
+  /**
+   * Creates a new account definition using the provided data. Secrets should use {@code
+   * UserSecretReference} encrypted URIs (e.g., {@code
+   * secret://secrets-manager?r=us-west-2&s=my-account-credentials}) when the underlying storage
+   * provider does not support row-level encryption or equivalent security features. Encrypted URIs
+   * will only be decrypted when loading account definitions, not when storing them. Note that
+   * account definitions correspond to the JSON representation of the underlying {@link
+   * CredentialsDefinition} object along with a JSON type discriminator field with the key {@code
+   * type} and value of the corresponding {@link CredentialsType} annotation.
+   *
+   * @param definition account definition to store as a new account
+   */
+  void create(CredentialsDefinition definition);
+
+  /**
+   * Creates or updates an account definition using the provided data. This is also known as an
+   * upsert operation. See {@link #create(CredentialsDefinition)} for more details.
+   *
+   * @param definition account definition to save
+   */
+  void save(CredentialsDefinition definition);
+
+  /** Creates or updates account definitions in bulk. */
+  void saveAll(Collection<CredentialsDefinition> definitions);
+
+  /**
+   * Updates an existing account definition using the provided data. See details in {@link
+   * #create(CredentialsDefinition)} for details on the format.
+   *
+   * @param definition updated account definition to replace an existing account
+   * @see #create(CredentialsDefinition)
+   */
+  void update(CredentialsDefinition definition);
+
+  /**
+   * Deletes an account by name.
+   *
+   * @param name name of account to delete
+   */
+  void delete(String name);
+
+  /**
+   * Deletes multiple accounts by name.
+   *
+   * @param names names of accounts to delete
+   */
+  void deleteAll(Collection<String> names);
+
+  /**
+   * Looks up the revision history of an account given its name. Revisions are sorted by latest
+   * version first.
+   *
+   * @param name account name to look up history for
+   * @return history of account updates for the given account name
+   */
+  List<Revision> revisionHistory(String name);
+
+  /**
+   * Provides metadata for an account definition revision when making updates to an account via
+   * {@link CredentialsDefinitionRepository} APIs.
+   */
+  class Revision {
+    private final int version;
+    private final long timestamp;
```

Comment:
I realize this is the same as in [clouddriver](https://github.com/spinnaker/clouddriver/blob/8ddccd8c21d1c06b22a910f1e44ef9c826fd6526/clouddriver-api/src/main/java/com/netflix/spinnaker/clouddriver/security/AccountDefinitionRepository.java#L119), but...if this won't bust already-serialized info, how 'bout making this all nice and explicit like:
```suggestion
    private final long timestampMillis;
```

Reply:
Kind of already being used in the UI.